crates/p2p/src/network.rs

```diff
@@ -0,0 +1,389 @@
+                    source_peer: propagation_source,
+                    id: message_id,
+                    message,
+                })
+            }
+            gossipsub::Event::Subscribed { peer_id, topic } => {
+                debug!("Peer {:?} subscribed to {:?}", peer_id, topic);
+                Some(NetworkEvent::Subscribe { peer_id, topic })
+            }
+            msg => {
+                debug!("{msg:?}");
+                None
+            }
+        }
+    }
+
+    /// handle reqrep event
+    fn handle_reqrep_event(&self, event: request_response::Event) -> Option<NetworkEvent> {
+        match event {
+            request_response::Event::Request {
+                peer_id,
+                request,
+                response_sender,
+                ..
+            } => match request {
+                Request::Ping(_ping) => {
+                    let response = Response::Pong(Default::default());
+                    response_sender
+                        .send(response)
+                        .expect("channel should exist");
+                    None
+                }
+                _ => Some(NetworkEvent::RequestMessage {
+                    peer_id,
+                    request,
+                    response_sender,
+                }),
+            },
+            request_response::Event::Response {
+                peer_id, response, ..
+            } => Some(NetworkEvent::ResponseMessage { peer_id, response }),
+            _ => None,
+        }
+    }
+
+    // TODO: discovery peer connect
+    fn handle_discovery_event(&self, _event: discovery::DiscoveredPeers) -> Option<NetworkEvent> {
+        None
+    }
+
+    /// handle peer manager event
+    fn handler_peer_manager_event(&mut self, event: PeerManagerEvent) -> Option<NetworkEvent> {
+        match event {
+            PeerManagerEvent::Ping(peer) => {
+                // FIXME: seq number should be a counter
+                self.send_request(&peer, Request::Ping(Ping::new(1)));
+                None
+            }
+            PeerManagerEvent::PeerConnectedIncoming(peer)
+            | PeerManagerEvent::PeerConnectedOutgoing(peer) => {
+                self.swarm
+                    .behaviour_mut()
+                    .gossipsub
+                    .add_explicit_peer(&peer);
+
+                Some(NetworkEvent::PeerConnected(peer))
+            }
+            PeerManagerEvent::PeerDisconnected(_) => None,
+            PeerManagerEvent::DiscoverPeers(_) => None,
+        }
+    }
+
+    pub fn poll_network(&mut self, cx: &mut Context) -> Poll<NetworkEvent> {
+        let mut msg_to_publish = Vec::new();
+        for (chain, ep, waiting_to_publish_ch, _) in self.entrypoint_channels.iter_mut() {
+            while let Ok(Some((pub_userop, verified_block))) = waiting_to_publish_ch.try_next() {
+                info!("Got userop {pub_userop:?} from ep {ep:} verified in {verified_block:?} to publish to p2p network!");
```

Comment:
Let's add another info! log to when bundler receives the user operation from RPC and it's succesfully validated - it's easier to see everything works ok

for example, replace this one with info and add some more information https://github.com/silius-rs/silius/blob/main/crates/uopool/src/uopool.rs#L228